ui/app/components/ui/list-item/list-item.component.js

```diff
@@ -1,138 +1,57 @@
-import React, { Component } from 'react'
+import React from 'react'
 import PropTypes from 'prop-types'
-import Approve from '../icon/approve-icon.component'
-import Interaction from '../icon/interaction-icon.component'
-import Preloader from '../icon/preloader'
-import Send from '../icon/send-icon.component'
 import classnames from 'classnames'
 
-const SendIcon = () => (
-  <Send
-    className="list-item__icon"
-    size={28}
-    color="#2F80ED"
-  />
-)
+export default function ListItem ({ title, subtitle, onClick, subtitleStatus, children, titleIcon, Icon, iconColor, iconSize = 28, rightContent, className }) {
+  const primaryClassName = classnames('list-item', className)
 
-const InteractionIcon = () => (
-  <Interaction
-    className="list-item__icon"
-    size={28}
-    color="#2F80ED"
-  />
-)
-
-const ApproveIcon = () => (
-  <Approve
-    className="list-item__icon"
-    size={28}
-    color="#2F80ED"
-  />
-)
-
-const FailIcon = () => (
-  <Interaction
-    className="list-item__icon"
-    size={28}
-    color="#D73A49"
-  />
-)
-
-export default class ListItem extends Component {
-  static contextTypes = {
-    t: PropTypes.func,
-  }
-
-  static propTypes = {
-    className: PropTypes.string,
-    status: PropTypes.string,
-    title: PropTypes.string.isRequired,
-    subtitle: PropTypes.string,
-    children: PropTypes.node,
-    primaryCurrency: PropTypes.string,
-    secondaryCurrency: PropTypes.string,
-  }
-
-  static defaultProps = {
-    status: 'pending',
-  }
-
-  render () {
-    const {
-      className,
-      status,
-      title,
-      subtitle,
-      children,
-      primaryCurrency,
-      secondaryCurrency,
-    } = this.props
-    const { t } = this.context
-
-    const isApproved = status === 'approved'
-    const isUnapproved = status === 'unapproved'
-    const isPending = status === 'pending'
-    const isFailed = status === 'failed'
-
-    let icon = <InteractionIcon />
-    if (isApproved) {
-      icon = <ApproveIcon />
-    } else if (isPending) {
-      icon = <SendIcon />
-    } else if (isFailed) {
-      icon = <FailIcon />
-    }
-
-    let subtitleStatus = null
-    if (isUnapproved) {
-      subtitleStatus = (
-        <span><span className="list-item__status--unapproved">{t('unapproved')}</span> · </span>
-      )
-    } else if (isFailed) {
-      subtitleStatus = (
-        <span><span className="list-item__status--failed">{t('failed')}</span> · </span>
-      )
-    }
-
-    return (
-      <div className={className}>
+  return (
+    <div className={primaryClassName} onClick={onClick}>
+      {Icon && (
         <div className="list-item__col">
-          { icon }
+          <Icon
```

Comment:
Huh - this isn't a pattern I've seen often. Would it be more conventional to pass in a node or a render prop instead of the component function? 🤔 Then the icon colour and size could be omitted as well.

I'm not totally against this as-written, but it feels a little strange to pass in a class plus the props, and asking the `ListItem` component to put them together.

Reply:
Ah, but the `list-item__icon` class name is added. Adding that to a `node` would require cloning, which is nice to avoid. This is definitely better than that option at least.

Reply:
Upon reflection I think this approach seems fine, but I'd still be curious to hear your thoughts on this vs. using a render prop.

Reply:
I like the render prop. We could make it a part of the API that className is one of the values received? ex:

```js
<ListItem icon={({ className }) => <SendIcon className={className} />} />
```

Downside is that if they don't do this, then the styles applied by the ListItem for icon is messed up.

Alternatively I could try to move the list-item__icon styles up a level in a containing element... then move it to a node type.

Reply:
```scss
  &__icon > * {
    margin: 8px 14px 0 0;
  }
```

add the __icon class to the div containing the icon element along with __col, then target any direct child of __icon. * could easily be svg for my purposes but i don't want to assume all icons will be svg

Reply:
@Gudahtt added a second commit that shows the changes made to remove the `Icon` component prop for `icon` node prop.

Reply:
>Downside is that if they don't do this, then the styles applied by the ListItem for icon is messed up.

A similar assumption is already being made about the component, as it might be something that doesn't accept `className`.

Though the `&__icon > *` approach seems better to me. Simpler to use.

Reply:
true dat